operator/pkg/ciliumidentity/cache_test.go

```diff
@@ -323,3 +329,72 @@ func TestCIDUsageInCES(t *testing.T) {
 	assert.Equal(t, 0, state.CIDUsageCount("1000"), assertTxt)
 	assert.Equal(t, 0, state.CIDUsageCount("2000"), assertTxt)
 }
+
+func TestEnqueueTimeTracker(t *testing.T) {
+	testCases := []struct {
+		name         string
+		item         string
+		setBeforeGet bool
+	}{
+		{"Set and get", "item1", true},
+		{"Get without set", "item2", false},
+	}
```

Comment:
I'm not sure that a table driven tests approach is the best here. Considering that we only have two subtests and each is slightly different (hence the `setBeforeGet` flag) I suggest to write a simple test without the multiple cases.

Reply:
I feel like it makes more sense for this one to be tested together, though I've renamed the setBeforeGet to increase the readability.

Reply:
I feel like it makes more sense for this one to be tested together, though I've renamed the setBeforeGet to increase the readability.

Reply:
What I meant was to keep those two cases in the same test, but avoid the table driven tests approach. What do you think about:

```
func TestEnqueueTimeTracker(t *testing.T) {
	tracker := &EnqueueTimeTracker{enqueuedAt: make(map[string]time.Time)}

	// test GetEnqueueTimeAndReset after initial set
	tracker.SetEnqueueTimeIfNotSet("item1")

	_, exists := tracker.GetEnqueueTimeAndReset("item1")
	if !exists {
		t.Error("Expected first GetEnqueueTimeAndReset(\"item1\") to return exists == true, got false")
	}

	_, exists = tracker.GetEnqueueTimeAndReset("item1")
	if exists {
		t.Error("Expected second GetEnqueueTimeAndReset(\"item1\") to return exists == false, got true")
	}

	// test GetEnqueueTimeAndReset without a previous set
	tracker.SetEnqueueTimeIfNotSet("item2")

	_, exists = tracker.GetEnqueueTimeAndReset("item2")
	if !exists {
		t.Error("Expected GetEnqueueTimeAndReset(\"item2\") to return exists == false, got true")
	}
}
```

I think this looks cleaner and also avoid calling two times `GetEnqueueTimeAndReset` in the second case ("without a previous set") where it is not necessary.

Reply:
What do you think about @sypakine proposal with `enqueuedItems` + `expect` pattern? I've updated my implementation to hopefully reflect this.

Reply:
Yep, this is better, thanks!